aiida_mlip/calculations/singlepoint.py

```diff
@@ -0,0 +1,226 @@
+        spec.output(
+            "results_dict",
+            valid_type=Dict,
+            help="The `results_dict` output node of the successful calculation.",
+        )
+        print("defining outputs")
+        spec.output("log_output", valid_type=SinglefileData)
+        spec.output("xyz_output", valid_type=SinglefileData)
+        print("defining outputnode")
+        spec.default_output_node = "results_dict"
+
+        # Exit codes
+        spec.exit_code(300, "INPUT_ERROR", message="Some problems reading the input")
+
+        spec.exit_code(
+            340,
+            "ERROR_OUT_OF_WALLTIME_INTERRUPTED",
+            message="The calculation stopped prematurely because it ran out of walltime",
+        )
+        spec.exit_code(
+            305,
+            "ERROR_MISSING_OUTPUT_FILES",
+            message="Some output files missing or cannot be read",
+        )
+        spec.exit_code(
+            306,
+            "ERROR_EMPTY_OUTPUT_FILES",
+            message="File.xyz is empty",
+        )
+        spec.exit_code(
+            307,
+            "ERROR_CONTENT_OUTPUT_FILES",
+            message="The output file does not contain the right content",
+        )
+
+    @classmethod
+    def validate_inputs(cls, value: dict) -> str:  # Need to work on this
+        """Check if the inputs are valid.
+
+        Parameters
+        ----------
+        value : dict
+            The inputs dictionary.
+
+        Returns
+        -------
+        str or None
+            Error message if validation fails, None otherwise.
+        """
+
+        # Check if  structure is given as it is required
+        for key in "structure":
+            if key not in value:
+                return f"required value was not provided for the `{key}` namespace."
+
+        return None
+
+    # pylint: disable=too-many-locals
+    def prepare_for_submission(
+        self, folder: aiida.common.folders.Folder
+    ) -> datastructures.CalcInfo:
+        """
+        Create the input files from the input nodes passed to this instance of the `CalcJob`.
+
+        Parameters
+        ----------
+        folder : aiida.common.folders.Folder
+            An `aiida.common.folders.Folder` to temporarily write files on disk.
+
+        Returns
+        -------
+        aiida.common.datastructures.CalcInfo
+            An instance of `aiida.common.datastructures.CalcInfo`.
+        """
+        # Create needed inputs
+        # Define architecture from model if model is given, otherwise get architecture from inputs and download default model
+        architecture = (
+            str((self.inputs.model).architecture)
+            if self.inputs.model
+            else str(self.inputs.architecture.value)
+        )
+        model_path = (
+            str((self.inputs.model).filepath)
+            if self.inputs.model
+            else ModelData.download(
+                "http://tinyurl.com/46jrkm3v", architecture
```

Comment:
I do not like seeing a random tinyurl link downloaded automatically. This is not safe.

Reply:
fair enough, changed with the model in janus-core